### ISSUE TITLE
SFX Implementation

### ISSUE THREAD

**Issue by HoracioYago** [Mathys-Hymon/GhostnGoblinsRemake#653]
## 🚀 Description

Implement SFX into most features within the game.

## 🔍 Blueprints used

**List the scenes or blueprints you use**. 
listing the maps, scenes and bp used prevents other people from touching them in parallel and generating corruption.

## 🎯 Purpose of the feature

Give signs, feedback, and make the game feel juicier.

## 📝 Expected features

ALL SFX in the Data Asset as arrays, play a random one within the array.
Mathys if anything is not understandable lmk and I'll explain.

**_Player_** 

Steps (Already Done)
Armor Jingle (Already Done)

On Jump 
On Landing
On Throw
On Take Damage (Armor)
On Take Damage (Voice)
On Death (Voice)
On Dash (Voice)
On Ultimate (Voice)

On Ult Bar Full 

Dissolve Fx (On player death)
Bones (After the dissolve, synched with the bone drop)

Death Screen (After skeleton drops)
Respawn (When respawning)

Dash (Fx)

Ultimate (Fx)
Get Armor Back (At the end of Ult)

**_Dagger_**

Charging (One sound per X dagger charged)
Full Charge (One sound)
Regular Shot 
Hit Zombie
Hit Ghost
Miss Hit (Floor, anything thats not an enemy)

**_Fireball_**

Charging 
Full Charge
Regular Shot
Charge Shot
Hit
Charge Hit
Cinders (Falling Fire)
Burning area (Loop)
Burning area turns off

**_Lance_**

Charging (Plays as long as player charges, otherwise interrupted)
Regular Shot
Charge Shot
Hit Zombie
Charge Hit Zombie
Hit Ghost
Charge Hit Ghost
Miss Hit (Floor, anything thats not an enemy)
Charge Miss Hit (Floor, anything thats not an enemy)

**_Zombie_**

Spawn (Voice)
Spawn Fx
Idle (When in idle)
Aggro (On detect Player)
Run (While Chasing)
Death (Voice)
Dissolve Fx (After death fx)

**_Ghost_** 

Spawn
Flying (Loop while Flying)
Breathe In (On animation before attack)
Attack (Shoot)
Dash (On dash before attack)
Hit (Take dmg)
Death (Voice)
Dissolve Fx (After death fx)

**_Bishop_**

Appear (Start Fight)
Flying (Loop Moving / Idle)
Go Invisible (When he goes invisible)
Breathe in (On animation before attack)
Cone Attack 
Spray Attack
Dash (On Dash before attack)
Clone Attack (When Cloning starts)
Clone Death (When Clone is destroyed)
Hit (When Hit)
Death (Voice)
Dissolve Fx (After death fx)

**_Enemy Fireball Fx_**

Launch (When appears)
Flight (Loop while flying)
Hit (On collision)

**_Souls_**

Souls collect (Soul collides with player)
Souls pop (Soul spawned)

**_Weapon Jar_**

Weapon Jar break (On being broken)
Weapon Aura (Loop Weapon drop GPE)
Weapon Equip  (On touching weapon drop = swap weapon)

**_Skull Doors / Tower Cage_**

Door skull lights up (Skull lights up on enemy death)
Door Opened / Tower Cage Hitbox Dispelled (When the required enemies are dead)
Door Dissolve (Dissolve Fx at the end)

**_Wisp_**

Talk (On player enter dialogue hitbox)
